radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/InMemoryAtomStore.java

```diff
@@ -179,9 +214,9 @@ public void store(RadixAddress address, AtomObservation atomObservation) {
 
 
 	@Override
-	public Stream<Particle> getUpParticles(RadixAddress address) {
+	public Stream<Particle> getUpParticles(RadixAddress address, String stagedUuid) {
 		synchronized (lock) {
-			return particleIndex.entrySet().stream()
+			Set<Particle> upParticles = particleIndex.entrySet().stream()
```

Comment:
It seems potentially dangerous that this does not preserve ordering - what's the reasoning behind this?

Reply:
Documenting this is probably enough - the returned Stream would indicate *to me* that it's ordered in some meaningful way

Reply:
Not sure what you mean, streams don't imply ordering they can even be parallel